quickstarts/microsoft.web/function-app-storage-private-endpoints/main.bicep

```diff
@@ -0,0 +1,442 @@
+      {
+        name: 'config'
+        properties: {
+          privateDnsZoneId: storageQueueDnsZone.id
+        }
+      }
+    ]
+  }
+}
+
+// -- Private Endpoints --
+resource storageFilePrivateEndpoint 'Microsoft.Network/privateEndpoints@2021-02-01' = {
+  name: privateEndpointStorageFileName
+  location: location
+  properties: {
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, privateEndpointSubnetName)
+    }
+    privateLinkServiceConnections: [
+      {
+        name: 'MyStorageFilePrivateLinkConnection'
+        properties: {
+          privateLinkServiceId: storageAccount.id
+          groupIds: [
+            'file'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageTablePrivateEndpoint 'Microsoft.Network/privateEndpoints@2021-02-01' = {
+  name: privateEndpointStorageTableName
+  location: location
+  properties: {
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, privateEndpointSubnetName)
+    }
+    privateLinkServiceConnections: [
+      {
+        name: 'MyStorageTablePrivateLinkConnection'
+        properties: {
+          privateLinkServiceId: storageAccount.id
+          groupIds: [
+            'table'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageQueuePrivateEndpoint 'Microsoft.Network/privateEndpoints@2021-02-01' = {
+  name: privateEndpointStorageQueueName
+  location: location
+  properties: {
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, privateEndpointSubnetName)
+    }
+    privateLinkServiceConnections: [
+      {
+        name: 'MyStorageQueuePrivateLinkConnection'
+        properties: {
+          privateLinkServiceId: storageAccount.id
+          groupIds: [
+            'queue'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageBlobPrivateEndpoint 'Microsoft.Network/privateEndpoints@2021-02-01' = {
+  name: privateEndpointStorageBlobName
+  location: location
+  properties: {
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, privateEndpointSubnetName)
+    }
+    privateLinkServiceConnections: [
+      {
+        name: 'MyStorageBlobPrivateLinkConnection'
+        properties: {
+          privateLinkServiceId: storageAccount.id
+          groupIds: [
+            'blob'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-02-01' = {
+  name: functionStorageAccountName
+  location: location
+  kind: 'StorageV2'
+  sku: {
+    name: 'Standard_LRS'
+  }
+  properties: {
+    networkAcls: {
+      bypass: 'None'
+      defaultAction: 'Deny'
+    }
+  }
+  dependsOn: [
+    virtualNetwork
+  ]
+}
+
+resource functionContentShare 'Microsoft.Storage/storageAccounts/fileServices/shares@2021-04-01' = {
+  name: '${storageAccount.name}/default/${functionContentShareName}'
+  dependsOn: [
```

Comment:
use the parent property and remove the dependsOn

Reply:
I removed 'dependsOn'.  I don't think `parent` is needed either.  The  **azuredeploy.json** file, resulting from `az bicep build -f` shows the dependency gets added to the ARM.

![image](https://user-images.githubusercontent.com/1142273/126670044-244653e9-dcad-4002-8eeb-9abf2bf79388.png)

Reply:
LGTM